Camera2Basic/utils/src/main/java/com/example/android/camera/utils/YuvToRgbConverter.kt

```diff
@@ -59,33 +58,30 @@ class YuvToRgbConverter(context: Context) {
             // Bits per pixel is an average for the whole image, so it's useful to compute the size
             // of the full buffer but should not be used to determine pixel offsets
             val pixelSizeBits = ImageFormat.getBitsPerPixel(ImageFormat.YUV_420_888)
-            yuvBuffer = ByteBuffer.allocateDirect(pixelCount * pixelSizeBits / 8)
+            yuvBuffer = ByteArray(pixelCount * pixelSizeBits / 8)
         }
 
-        // Rewind the buffer; no need to clear it since it will be filled
-        yuvBuffer.rewind()
-
         // Get the YUV data in byte array form using NV21 format
-        imageToByteBuffer(image, yuvBuffer.array())
```

Comment:
This was a dangerous line: direct ByteBuffer is not required to be backed by array!

Reply:
Good point. 

For others: see [here](https://developer.android.com/reference/java/nio/ByteBuffer.html#array()) for exceptions this method can throw.